main/index.js

```diff
@@ -82,6 +87,18 @@ const ctx = {
   loadWebUIFromDist: serve({ directory: path.resolve(__dirname, '../renderer/dist') })
 }
 
+app.on('before-quit', () => {
+  // Flush pending events immediately
+  // See https://docs.sentry.io/platforms/node/configuration/draining/
+  Sentry.close()
+})
+
+process.on('uncaughtException', err => {
+  Sentry.captureException(err)
+  log.error(err)
+  process.exitCode = 1
```

Comment:
will this still close the process on uncaught exception?

Reply:
I am glad you have asked!

I think it depends. When the uncaught exception was created from an unhandled promise rejection, then the process keeps running. If it's an error thrown from a callback, then the app exits.

The current behaviour after an unhandled error/rejection is a bit fuzzy at the moment. I was hoping that my current change was not making the situation any worse, but I could be wrong.

Maybe this is the right time to revisit our error-handling strategy. What would you consider the right solution?

Note that for errors thrown from our setup functions, we don't exit the app.
https://github.com/filecoin-project/filecoin-station/blob/2e38e120688dbbf8f66142f9321d2ac4af61bdfd/main/index.js#L29-L38

Reply:
To me the main concern is for this PR not to change any behavior by accident. If the app was already not quitting on uncaught exceptions, then +1 to keeping this logic, until we decide to change it. If you're confirming that the logic basically is still the same, then I'm +1 with this change

Reply:
See https://github.com/filecoin-project/filecoin-station/pull/184#issuecomment-1262350907